### DISCUSSION TITLE
is there any way to get clientId and clientSecret from a async function per each singin request

### DISCUSSION THREAD

**Discussion by feyzullahyildiz** [nextauthjs/next-auth#10920] [Help] [answered]
I do have a super-admin page for only super-admins. They can update clientId and clientSecret at the runtime. We planned to store these values in our database.

I want to use new clientId and clientSecret values after the updates. I cannot do that right now.

```js

import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";
import db from "@db";

export const authOptions = {
  providers: [
    GoogleProvider({
      name: "google",
      clientId: async () => {
        const settings = await db.Settings.findOne();
        return settings.googleAuth.clientId
      },
      clientSecret: async () => {
        const settings = await db.Settings.findOne();
        return settings.googleAuth.clientSecret
      },


```

**Comment by k-taro56** [comment, marked as answer]
Hi,

next-auth does not automatically update database values.
If you need to update values after sign-in time, you must update the database manually.

Thank you.
